src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -295,9 +390,11 @@ private AIState cookFood()
         {
             walkTo = null;
             needsCurrently = ItemStackUtils.ISCOOKABLE;
-            return COOK_GATHERING;
+            return COOK_GATHER_FOOD_FROM_BUILDING;
         }
 
+        // TODO: this could leave us walking to a full furnace
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)